### PR TITLE
Added sample for WinAppSDK CameraCaptureUI

### PR DESCRIPTION
**Pull request by Saharsh979**
## Description
This pull request adds a sample application demonstrating the use of the CameraCaptureUI API in WinAppSDK. The sample showcases how to capture both photos and videos using custom settings such as format and resolution. It includes implementations in C++ and C# to highlight cross-language compatibility. The sample aims to provide developers with a practical example of integrating camera functionality into desktop apps.

## Target Release
1.7

## Checklist

- [x] Samples build and run using the Visual Studio versions listed in the [Windows development docs](https://docs.microsoft.com/windows/apps/windows-app-sdk/set-up-your-development-environment?tabs=stable#2-install-visual-studio).
- [x] Samples build and run on all supported platforms (x64, x86, ARM64) and configurations (Debug, Release).
- [x] Samples set the minimum supported OS version to Windows 10 version 1809.
- [x] Samples build clean with no warnings or errors.
- [x] **[For new samples]**: Samples have completed the [sample guidelines checklist](https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/docs/samples-guidelines.md#checklist) and follow [standardization/naming guidelines](https://github.com/microsoft/WindowsAppSDK-Samples/blob/main/docs/samples-guidelines.md#standardization-and-naming).

### CODE REVIEW

File: Samples/CameraCaptureUI/cpp-winui/cpp-winui/MainPage.xaml.cpp [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+        // If called from the UI thread, then update immediately.
+        // Otherwise, schedule a task on the UI thread to perform the update.
+        if (this->DispatcherQueue().HasThreadAccess())
+        {
+            UpdateStatus(strMessage, severity);
+        }
+        else
+        {
+            DispatcherQueue().TryEnqueue([strongThis = get_strong(), this, strMessage, severity]
+                { UpdateStatus(strMessage, severity); });
```
Comment (godlytalias) [inline Samples/CameraCaptureUI/cpp-winui/cpp-winui/MainPage.xaml.cpp:41]: Here the references of `strMessage` & `severity` is getting passed, It won't ensure that at the time UpdateStatus gets executed in UI Thread these memory will still hold the expected values as the notification will be over by then and the memory can get released.

File: Samples/CameraCaptureUI/cpp-winui/cpp-winui/pch.h [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#include <winrt/Microsoft.UI.Xaml.Shapes.h>
+#include <winrt/Microsoft.UI.Dispatching.h>
+#include <winrt/Windows.UI.Core.h>
+#include "winrt/Microsoft.Windows.Media.Capture.h"
+#include "winrt/Windows.Storage.h"
+#include "winrt/Microsoft.UI.Windowing.h"
+#include "winrt/Windows.ApplicationModel.h"
+#include "Winrt/Microsoft.UI.Xaml.Media.Imaging.h"
+#include <winrt/Windows.Storage.Pickers.h>
+#include <winrt/Windows.Storage.h>
```
Comment (godlytalias) [inline Samples/CameraCaptureUI/cpp-winui/cpp-winui/pch.h:38]: Some of the headers seems to be repeated

**Comment by Saharsh979** [comment]
Skipping setting up the actual pipeline for now because the "ms" ADO has been deprecated and we currently don't have a new home for these samples validation pipeline yet.

File: Samples/CameraCaptureUI/cpp-winui/cpp-winui/MainPage.xaml.cpp (hunk 1 above)
Comment (Saharsh979) [reply Samples/CameraCaptureUI/cpp-winui/cpp-winui/MainPage.xaml.cpp:41]: This is the code that came as part of the sample template I used to create the sample.
I can see that other samples created using the template are using this implementation without any modifications.

File: Samples/CameraCaptureUI/cpp-winui/cpp-winui/pch.h (hunk 2 above)
Comment (Saharsh979) [reply Samples/CameraCaptureUI/cpp-winui/cpp-winui/pch.h:36]: Removed, Thanks!

File: Samples/CameraCaptureUI/cpp-winui/cpp-winui/Scenario2_CaptureVideo.xaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+                    Foreground ="Red"
+                    FontSize="16"
+                    Width="200"
+                    Height="50"
+                    BorderThickness="2"
+                    CornerRadius="25"/>
+
+                <!-- Media Element: This is where the captured video will be displayed -->
+                <MediaPlayerElement
+                    x:Name="CapturedVideo"
```
Comment (Hemantxk) [inline Samples/CameraCaptureUI/cpp-winui/cpp-winui/Scenario2_CaptureVideo.xaml:48]: Rename this to mediaPlayElement for clarity?

File: Samples/CameraCaptureUI/cswinui/cswinui/Scenario2_CaptureVideo.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                var mediaSource = MediaSource.CreateFromStream(fileStream, file.ContentType);
+
+                // Create a MediaPlayer to play the video
+                var mediaPlayer = new MediaPlayer();
+
+                // Set the media source for the MediaPlayer
+                mediaPlayer.Source = mediaSource;
+
+                // Set the MediaPlayer to the MediaPlayerElement control in the UI
+                CapturedVideo.SetMediaPlayer(mediaPlayer);
```
Comment (Hemantxk) [inline Samples/CameraCaptureUI/cswinui/cswinui/Scenario2_CaptureVideo.xaml.cs:70]: CapturedVideo is a misleading name that represents a MediaPlayerElement

File: Samples/CameraCaptureUI/cswinui/cswinui/SampleConfiguration.cs [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Controls;
+using System.Collections.Generic;
+
+namespace cswinui
+{
+    internal static class SampleConfig
+    {
+        public const string FeatureName = "cswinui";
```
[the remark this answers is not in the document]
Comment (Saharsh979) [reply Samples/CameraCaptureUI/cswinui/cswinui/SampleConfiguration.cs]: Updated!

File: Samples/CameraCaptureUI/cswinui/cswinui/Scenario2_CaptureVideo.xaml.cs (hunk 4 above)
Comment (Saharsh979) [reply Samples/CameraCaptureUI/cswinui/cswinui/Scenario2_CaptureVideo.xaml.cs]: Makes sense, updated!

**Review by Hemantxk — APPROVED** [review]

**Review by godlytalias — APPROVED** [review]

**Pull request merged**